src/librustc/diagnostics.rs

```diff
@@ -1431,6 +1431,21 @@ fn make_child<'elve>(x: &mut &'elve isize, y: &mut &'elve isize) {
 ```
 "##,
 
+E0317: r##"
+An `if` expression without an `else` block is required to have the type `()`.
+This error occurs when the `if` block has a type other than `()`. For example:
+
+```compile_fail,E0317
+fn main() {
+    let x = 5;
+    let a = if x == 5 { 1 };
```

Comment:
Does it? That’s weird. I would expect `else`-less `if` as a last expression in a function body/block to produce the “if may be missing an else clause” error instead.